equinix/resource_fabric_connection.go

```diff
@@ -0,0 +1,265 @@
+	project := projectToFabric(projectReq)
+	connectionASide := v4.ConnectionSide{}
+	for _, as := range aside {
+		asideMap := as.(map[string]interface{})
+		accessPoint := asideMap["access_point"].(interface{}).(*schema.Set).List()
+		serviceTokenRequest := asideMap["service_token"].(interface{}).(*schema.Set).List()
+		additionalInfoRequest := asideMap["additional_info"].([]interface{})
+
+		if len(accessPoint) != 0 {
+			ap := accessPointToFabric(accessPoint)
+			connectionASide = v4.ConnectionSide{AccessPoint: &ap}
+		}
+		if len(serviceTokenRequest) != 0 {
+			mappedServiceToken := serviceTokenToFabric(serviceTokenRequest)
+			connectionASide = v4.ConnectionSide{ServiceToken: &mappedServiceToken}
+		}
+		if len(additionalInfoRequest) != 0 {
+			mappedAdditionalInfo := additionalInfoToFabric(additionalInfoRequest)
+			connectionASide = v4.ConnectionSide{AdditionalInfo: mappedAdditionalInfo}
+		}
+	}
+
+	zside := d.Get("z_side").(interface{}).(*schema.Set).List()
+	connectionZSide := v4.ConnectionSide{}
+	for _, as := range zside {
+		zsideMap := as.(map[string]interface{})
+		accessPoint := zsideMap["access_point"].(interface{}).(*schema.Set).List()
+		serviceTokenRequest := zsideMap["service_token"].(interface{}).(*schema.Set).List()
+		additionalInfoRequest := zsideMap["additional_info"].([]interface{})
+		if len(accessPoint) != 0 {
+			ap := accessPointToFabric(accessPoint)
+			connectionZSide = v4.ConnectionSide{AccessPoint: &ap}
+		}
+		if len(serviceTokenRequest) != 0 {
+			mappedServiceToken := serviceTokenToFabric(serviceTokenRequest)
+			connectionZSide = v4.ConnectionSide{ServiceToken: &mappedServiceToken}
+		}
+		if len(additionalInfoRequest) != 0 {
+			mappedAdditionalInfo := additionalInfoToFabric(additionalInfoRequest)
+			connectionZSide = v4.ConnectionSide{AdditionalInfo: mappedAdditionalInfo}
+		}
+	}
+
+	createRequest := v4.ConnectionPostRequest{
+		Name:          d.Get("name").(string),
+		Type_:         &conType,
+		Order:         &order,
+		Notifications: notifications,
+		Bandwidth:     int32(d.Get("bandwidth").(int)),
+		Redundancy:    &red,
+		ASide:         &connectionASide,
+		ZSide:         &connectionZSide,
+		Project:       &project,
+	}
+
+	conn, _, err := client.ConnectionsApi.CreateConnection(ctx, createRequest)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	d.SetId(conn.Uuid)
+	return resourceFabricConnectionRead(ctx, d, meta)
+}
+
+func resourceFabricConnectionRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	log.Printf(" inside fabric connection read resource ")
+	client := meta.(*Config).fabricClient
+	ctx = context.WithValue(ctx, v4.ContextAccessToken, meta.(*Config).FabricAuthToken)
+	conn, _, err := client.ConnectionsApi.GetConnectionByUuid(ctx, d.Id(), nil)
+
+	if err != nil {
+		log.Printf("[WARN] Connection %s not found ", d.Id())
+		d.SetId("")
+		return diag.FromErr(err)
+	}
+	d.SetId(conn.Uuid)
```

Comment:
If the connection has a deleted status, Terraform should set `d.SetID("")` and `return nil`.

Reply:
As discussed, in Connection Management connections are soft delete and customers are allowed to fetch deprovisioned connections.